lib/src/main/java/uk/gov/di/ipv/cri/address/library/models/CanonicalAddress.java

```diff
@@ -1,27 +1,36 @@
 package uk.gov.di.ipv.cri.address.library.models;
 
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBDocument;
 import com.fasterxml.jackson.annotation.JsonInclude;
 import uk.gov.di.ipv.cri.address.library.models.ordnancesurvey.Result;
 
+import java.util.Optional;
+
 @JsonInclude(JsonInclude.Include.NON_NULL)
+@DynamoDBDocument
 public class CanonicalAddress {
-    private final String uprn;
-    private final String organisationName;
-    private final String departmentName;
-    private final String subBuildingName;
-    private final String buildingNumber;
-    private final String dependentThoroughfare;
-    private final String doubleDependentLocality;
-    private final String dependentLocality;
-    private final String buildingName;
-    private final String thoroughfareName;
-    private final String postTown;
-    private final String postcode;
-    private final String countryCode;
+    private Optional<Integer> uprn;
```

Comment:
I'm not sure this is idiomatic use of optional, I mean declaring fields as Optional rather I think the returning method could be optional instead, I don't know how this would affect the stuff you've done so maybe something like this is more idiomatic 

private Integer uprn;

 public Optional<Integer> getUprn() {
      return Optional.ofNullable(uprn);
 }

 public void setUprn(Integer uprn) {
      this.uprn = uprn;
  }